src/Microsoft.DotNet.Interactive.Jupyter/Http/JupyterHttpConnection.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Interactive.Jupyter.Connection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Reactive.Disposables;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive.Jupyter.Http;
+
+internal class JupyterHttpConnection : IJupyterConnection
+{
+    #region JsonTypes
+    private class KernelSessionInfo
+    {
+        public string id { get; set; }
+        public string path { get; set; }
+        public string name { get; set; }
+        public string type { get; set; }
+        public KernelInfo kernel { get; set; }
+    }
+
+    private class KernelInfo
+    {
+        public string id { get; set; }
+        public string name { get; set; }
+
+        public string last_activity { get; set; }
+        public string execution_state { get; set; }
+        public int connections { get; set; }
+    }
+
+    private class KernelSpecs
+    {
+        public string @default { get; set; }
+        public Dictionary<string, KernelSpecDetail> kernelspecs { get; set; }
+    }
+
+    private class KernelSpecDetail
+    {
+        public string name { get; set; }
+        public KernelSpec spec { get; set; }
+        public object resources { get; set; }
+    }
+    #endregion
+
+    private readonly HttpApiClient _apiClient;
+    private readonly IAuthorizationProvider _authProvider;
+    private readonly CompositeDisposable _disposables;
+    private IEnumerable<KernelSpec> _availableKernels;
+    private List<string> _activeSessions = new();
+
+    public JupyterHttpConnection(Uri serverUri, IAuthorizationProvider authProvider) :
+        this(new HttpApiClient(serverUri, authProvider), authProvider)
+    { }
+
+    public JupyterHttpConnection(HttpApiClient apiClient, IAuthorizationProvider authProvider)
+    {
+        _apiClient = apiClient ?? throw new ArgumentNullException(nameof(apiClient));
+        _authProvider = authProvider ?? throw new ArgumentNullException(nameof(authProvider));
+
+        _disposables = new CompositeDisposable
+        {
+            _apiClient
+        };
+    }
+
+    public void Dispose()
+    {
+        Task.Run(ShutdownJupyterSessionsAsync).Wait();
```

Comment:
We shouldn't call `Wait`. Consider using `IAsyncDisposable`.

Reply:
Will remove the Wait since it's already on a Task.Run. Will revisit the design.